packages/react-dom/src/server/ReactPartialRendererHooks.js

```diff
@@ -459,6 +463,15 @@ function useResponder(responder, props): ReactEventResponderListener<any, any> {
   };
 }
 
+function useMutableSource<Source, Snapshot>(
+  source: MutableSource<Source>,
+  config: MutableSourceHookConfig<Source, Snapshot>,
+): Snapshot {
+  resolveCurrentlyRenderingComponent();
+  const getSnapshot = config.getSnapshot;
+  return getSnapshot(source._source);
```

Comment:
I can’t think of a server use case that isn’t actually Suspense/Flight. So never mind.

Reply:
Hm. I think this is legit even in the existing server renderer. You can use a shared in memory cache that changes. E.g. another request might be invalidating it at the same time you're rendering.

I'm just not sure how to solve it since it requires a second pass. I think the best we could do is trigger Suspense boundaries and fallback to client rendering. That might be the TODO.

Reply:
I'll leave a TODO for now.